From e02616866fe2461836dc8f7f6e369b3b25644cc4 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@php.net>
Date: Thu, 21 Aug 2014 09:09:21 +0200
Subject: [PATCH] Fix bug #67878 program_prefix not honoured in man pages

- add @program_prefix@ in php, phpize and php-config man pages
- set date to 2014
- fix "SEE ALSO" alignment
---
 sapi/cli/php.1.in            | 34 +++++++++++++++++-----------------
 scripts/man1/php-config.1.in | 14 +++++++-------
 scripts/man1/phpize.1.in     | 14 +++++++-------
 3 files changed, 31 insertions(+), 31 deletions(-)

diff --git a/sapi/cli/php.1.in b/sapi/cli/php.1.in
index 4d536df..c062f3d 100644
--- a/sapi/cli/php.1.in
+++ b/sapi/cli/php.1.in
@@ -1,24 +1,24 @@
-.TH PHP 1 "2014" "The PHP Group" "Scripting Language"
+.TH @program_prefix@php 1 "2014" "The PHP Group" "Scripting Language"
 .SH NAME
-php \- PHP Command Line Interface 'CLI'
+@program_prefix@php \- PHP Command Line Interface 'CLI'
 .P
-php-cgi \- PHP Common Gateway Interface 'CGI' command
+@program_prefix@php-cgi \- PHP Common Gateway Interface 'CGI' command
 .SH SYNOPSIS
-.B php
+.B @program_prefix@php
 [options] [
 .B \-f\fP ]
 .IR file
 [[\-\-] 
 .IR args.\|.\|. ]
 .LP
-.B php
+.B @program_prefix@php
 [options] 
 .B \-r 
 .IR code
 [[\-\-]
 .IR args.\|.\|. ]
 .LP
-.B php
+.B @program_prefix@php
 [options] [\-B 
 .IR begin_code ]
 .B \-R 
@@ -28,7 +28,7 @@ php-cgi \- PHP Common Gateway Interface 'CGI' command
 [[\-\-]
 .IR args.\|.\|. ]
 .LP
-.B php
+.B @program_prefix@php
 [options] [\-B 
 .IR begin_code ]
 .B \-F 
@@ -38,13 +38,13 @@ php-cgi \- PHP Common Gateway Interface 'CGI' command
 [[\-\-]
 .IR args.\|.\|. ]
 .LP
-.B php
+.B @program_prefix@php
 [options] \-\- [
 .IR args.\|.\|. ]
 .LP
-\fBphp \fP[options] \fB\-a\fP
+\fB@program_prefix@php \fP[options] \fB\-a\fP
 .LP
-.B php
+.B @program_prefix@php
 [options] \-S
 .IR addr:port
 [\-t
@@ -374,35 +374,35 @@ The standard configuration file will only be used when
 cannot be found.
 .SH EXAMPLES
 .TP 5
-\fIphp \-r 'echo "Hello World\\n";'\fP
+\fI@program_prefix@php \-r 'echo "Hello World\\n";'\fP
 This command simply writes the text "Hello World" to standard out.
 .TP
-\fIphp \-r 'print_r(gd_info());'\fP
+\fI@program_prefix@php \-r 'print_r(gd_info());'\fP
 This shows the configuration of your gd extension. You can use this
 to easily check which image formats you can use. If you have any
 dynamic modules you may want to use the same ini file that php uses
 when executed from your webserver. There are more extensions which
 have such a function. For dba use:
 .RS
-\fIphp \-r 'print_r(dba_handlers(1));'\fP
+\fI@program_prefix@php \-r 'print_r(dba_handlers(1));'\fP
 .RE
 .TP
-\fIphp \-R 'echo strip_tags($argn)."\\n";'\fP
+\fI@program_prefix@php \-R 'echo strip_tags($argn)."\\n";'\fP
 This PHP command strips off the HTML tags line by line and outputs the 
 result. To see how it works you can first look at the following PHP command
 \'\fIphp \-d html_errors=1 \-i\fP\' which uses PHP to output HTML formatted
 configuration information. If you then combine those two 
 \'\fIphp \.\.\.|php \.\.\.\fP\' you'll see what happens.
 .TP
-\fIphp \-E 'echo "Lines: $argi\\n";'\fP
+\fI@program_prefix@php \-E 'echo "Lines: $argi\\n";'\fP
 Using this PHP command you can count the lines being input.
 .TP
-\fIphp \-R '@$l+=count(file($argn));' \-E 'echo "Lines:$l\\n";'\fP
+\fI@program_prefix@php \-R '@$l+=count(file($argn));' \-E 'echo "Lines:$l\\n";'\fP
 In this example PHP expects each input line being a file. It counts all lines 
 of the files specified by each input line and shows the summarized result. 
 You may combine this with tools like find and change the php scriptlet.
 .TP
-\fIphp \-R 'echo "$argn\\n"; fgets(STDIN);'\fP
+\fI@program_prefix@php \-R 'echo "$argn\\n"; fgets(STDIN);'\fP
 Since you have access to STDIN from within \-B \-R \-F and \-E you can skip 
 certain input lines with your code. But note that in such cases $argi only 
 counts the lines being processed by php itself. Having read this you will 
diff --git a/scripts/man1/php-config.1.in b/scripts/man1/php-config.1.in
index 5c60408..ae2ee56 100644
--- a/scripts/man1/php-config.1.in
+++ b/scripts/man1/php-config.1.in
@@ -1,12 +1,12 @@
-.TH php\-config 1 "2010" "The PHP Group" "Scripting Language"
+.TH @program_prefix@php\-config 1 "2014" "The PHP Group" "Scripting Language"
 .SH NAME
-php\-config \- get information about PHP configuration and compile options
+@program_prefix@php\-config \- get information about PHP configuration and compile options
 .SH SYNOPSIS
-.B php\-config
+.B @program_prefix@php\-config
 [options]
 .LP
 .SH DESCRIPTION
-.B php\-config
+.B @program_prefix@php\-config
 is a simple shell script for obtaining information about installed PHP configuration.
 .SH OPTIONS
 .TP 15
@@ -57,15 +57,15 @@ PHP version
 .PD 0
 .B \-\-vernum
 PHP version as integer
-.TP
+.RS
 .PD 1
 .P
 .SH SEE ALSO
-.BR php (1)
+.BR @program_prefix@php (1)
 .SH VERSION INFORMATION
 This manpage describes \fBphp\fP, version @PHP_VERSION@.
 .SH COPYRIGHT
-Copyright \(co 1997\-2010 The PHP Group
+Copyright \(co 1997\-2014 The PHP Group
 .LP
 This source file is subject to version 3.01 of the PHP license,
 that is bundled with this package in the file LICENSE, and is
diff --git a/scripts/man1/phpize.1.in b/scripts/man1/phpize.1.in
index 8f62f86..e49adb1 100644
--- a/scripts/man1/phpize.1.in
+++ b/scripts/man1/phpize.1.in
@@ -1,12 +1,12 @@
-.TH phpize 1 "2010" "The PHP Group" "Scripting Language"
+.TH @program_prefix@phpize 1 "2014" "The PHP Group" "Scripting Language"
 .SH NAME
-phpize \- prepare a PHP extension for compiling
+@program_prefix@phpize \- prepare a PHP extension for compiling
 .SH SYNOPSIS
-.B phpize
+.B @program_prefix@phpize
 [options]
 .LP
 .SH DESCRIPTION
-.B phpize
+.B @program_prefix@phpize
 is a shell script to prepare PHP extension for compiling.
 .SH OPTIONS
 .TP 15
@@ -24,15 +24,15 @@ Prints usage information
 .PD 1
 .B \-v
 Prints API version information
-.TP
+.RS
 .PD 1
 .P
 .SH SEE ALSO
-.BR php (1)
+.BR @program_prefix@php (1)
 .SH VERSION INFORMATION
 This manpage describes \fBphp\fP, version @PHP_VERSION@.
 .SH COPYRIGHT
-Copyright \(co 1997\-2010 The PHP Group
+Copyright \(co 1997\-2014 The PHP Group
 .LP
 This source file is subject to version 3.01 of the PHP license,
 that is bundled with this package in the file LICENSE, and is
-- 
1.9.2

